### DISCUSSION TITLE
Ads are not shown

### DISCUSSION THREAD

**Discussion by rajshah5599** [community/community#130393] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Testing ads are working in a test environment, but live ads are not showing. I have been waiting for 2 weeks, but they are still not showing in my React Native app.

**Comment by CodeCandyCelvin** [comment]
There could be several reasons why your ads are not showing in the live environment even though they work in the test environment. Here are some common issues and steps you can take to troubleshoot:

Ad Unit IDs: Ensure you are using the correct ad unit IDs for the live environment. Test ads use different ad unit IDs compared to live ads. Double-check that the IDs you are using in production are correct and active.

App Approval: Verify that your app has been approved for ad serving by the ad network (e.g., Google AdMob). Some ad networks require your app to go through an approval process before serving live ads.

Ad Inventory Availability: Sometimes, the ad network may not have available inventory to serve ads to your users. This can be affected by factors such as the location of your users or the targeting criteria set for your ads.

Ad Network Policy Compliance: Ensure that your app complies with all ad network policies. If your app violates any policies, ads may be blocked from being shown.

Ad Loading Code: Check your ad loading implementation to ensure there are no errors or issues in your code. Make sure the ad requests are being made correctly and that there are no issues with the ad callbacks.

Logs and Error Messages: Review the logs and error messages in your app to identify any issues related to ad loading. Look for any error codes or messages that might indicate the problem.

Ad Network SDK: Ensure that you are using the latest version of the ad network's SDK. Outdated SDK versions might have bugs or compatibility issues that prevent ads from showing.

Network Connectivity: Ensure that your app has a stable internet connection. Ads cannot be served if there is no internet connectivity or if the connection is unstable.

User Privacy Settings: Verify that your app’s privacy settings are configured correctly and that users have given consent to receive ads, especially in regions with strict privacy regulations like GDPR.
